source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -481,90 +683,100 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
         // Start traversal from this output
         root = output;
     }
+
     else if (element->isA<Node>())
     {
         NodePtr node = element->asA<Node>();
 
-        NodeDefPtr nodeDef = node->getNodeDef();
-        if (!nodeDef)
+        // Handle shader nodes different from other nodes
+        if (node->getType() == SURFACE_SHADER_TYPE_STRING)
```

Comment:
The code after this is messed up in the compare. There are no changes to that code. It just got put into the "else" clause when it's not a shader.